cpp/Contact.cpp

```diff
@@ -413,9 +413,9 @@ void dolfinx_contact::Contact::assemble_vector(
   std::shared_ptr<const dolfinx::graph::AdjacencyList<int>> map
       = _facet_maps[pair];
   std::shared_ptr<const dolfinx::graph::AdjacencyList<int>> facet_map
-      = _submeshes[_contact_pairs[pair][1]].facet_map();
+      = _submeshes[contact_pair[1]].facet_map();
```

Comment:
Seems like `_submeshes[contact_pair[1]]` should have a temporary `const dolfinx_contact::SubMesh& submesh = _submeshes[contact_pair[1]]` assignment